server/src/test/java/org/elasticsearch/ReleaseVersionsTests.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch;
+
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.function.IntFunction;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class ReleaseVersionsTests extends ESTestCase {
+
+    public void testReleaseVersions() {
+        IntFunction<String> versions = ReleaseVersions.generateVersionsLookup(ReleaseVersionsTests.class);
+
+        assertThat(versions.apply(10), equalTo("8.0.0"));
+        assertThat(versions.apply(14), equalTo("8.1.0-8.1.1"));
+        assertThat(versions.apply(21), equalTo("8.2.0"));
+        assertThat(versions.apply(22), equalTo("8.2.1"));
+    }
+
+    public void testReturnsRange() {
+        IntFunction<String> versions = ReleaseVersions.generateVersionsLookup(ReleaseVersionsTests.class);
+
+        assertThat(versions.apply(17), equalTo("8.1.2-8.2.0"));
+        expectThrows(AssertionError.class, () -> versions.apply(9));
+        assertThat(versions.apply(24), equalTo("8.2.2-[24]"));
```

Comment:
This would be confusing in a log message. Could we change this to omit the upper bound, ie `8.2.2-`?

Reply:
I think just having a trailing `-` would be a bit confusing. This should only be seen in snapshot builds used by devs, so having the id displayed is useful for diagnose the problem the log/exception message is for. Is there a different format you think we could use?

Reply:
If this is only seen by "us" (devs) I think it is useful for debugging

Reply:
If it's only for devs, then perhaps being more explicit? I don't think devs will immediately understand this bracket notation (which will never be seen in production) is special and is a "raw" value.

Reply:
How about something like `8.12.0-snapshot[8546000]`?

Reply:
That sounds good to me.